From dc46cfd787f753969d7e4afb97d6bc234896fdaf Mon Sep 17 00:00:00 2001
From: Daniel Chavolla <daniel.chavolla@windriver.com>
Date: Thu, 18 May 2017 09:52:26 -0400
Subject: [PATCH] add error reporting when resizing to a smaller disk

When resizing a booted from image VM, we already check for the
"resizing from a non-zero root_gb to a zero root_gb" case. Given
that resizing to a smaller disk is ultimately not allowed and
rejected by the libvirt driver, make sense to extend the check here
to prevent that case. Note that since this error is now being
caught early, it is correctly propagated to Heat.
---
 nova/compute/api.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/nova/compute/api.py b/nova/compute/api.py
index 794025a..b18f0ce 100644
--- a/nova/compute/api.py
+++ b/nova/compute/api.py
@@ -3151,11 +3151,11 @@ class API(base.Base):
         else:
             new_instance_type = flavors.get_flavor_by_flavor_id(
                     flavor_id, read_deleted="no")
-            if (new_instance_type.get('root_gb') == 0 and
-                current_instance_type.get('root_gb') != 0 and
+            if (new_instance_type.get('root_gb') <
+                current_instance_type.get('root_gb') and
                 not compute_utils.is_volume_backed_instance(context,
                     instance)):
-                reason = _('Resize to zero disk flavor is not allowed.')
+                reason = _('Resize to smaller disk flavor is not allowed.')
                 raise exception.CannotResizeDisk(reason=reason)
 
             if (new_instance_type.get('ephemeral_gb') <
-- 
2.7.4

